src/crawler.js

```diff
@@ -61,7 +65,7 @@ function normalizeIpp(ip, port) {
 
     out_ip = splitIp
     out_port = port || splitPort || DEFAULT_PORT
-    ipp = out_ip + ':' + out_port
+    var ipp = out_ip + ':' + out_port
   }
 
   return ipp
```

Comment:
if ip is undefined, isn't it expected for this function to return undefined also?
it's up to the caller to not pass in an undefined ip